Added stop sequence parameter to ChatBedrock model

zhaoJoseph opened the pull request:
This is a fix for https://github.com/langchain-ai/langchain-aws/issues/234. Test still passes while noting it is not implemented.

michaelnchin reviewed (changes requested):
@zhaoJoseph, thank you for taking a look at this.

We need to also be providing the value of `stop_sequences` to the model to make the parameter functional.

Can you add additional code to ensure that the value of `stop_sequences` is passed with `inferenceConfig` in the [body of the invoke_model request](https://docs.aws.amazon.com/bedrock/latest/APIReference/API_runtime_InvokeModel.html)?

The implementation in `ChatBedrockConverse` class can be used as a reference:
https://github.com/langchain-ai/langchain-aws/blob/main/libs/aws/langchain_aws/chat_models/bedrock_converse.py#L632

zhaoJoseph commented:
Hi @michaelnchin, thanks for reviewing my PR. I have taken a look at ``ChatBedrockConverse`` class and added an implementation for ``stop_sequences`` in ``ChatBedrock``. Also modified the test case for  ``stop_sequences`` to reflect this change.

michaelnchin commented on libs/aws/tests/integration_tests/chat_models/test_standard.py:
```diff
@@ -44,6 +39,7 @@ def standard_chat_model_params(self) -> dict:
         return {
             "temperature": 0,
             "max_tokens": 100,
+            "stop_sequences": [],
```
We shouldn't include `stop_sequences` here, as it's not a standard ChatModel parameter: https://python.langchain.com/docs/concepts/chat_models/#standard-parameters

michaelnchin commented on libs/aws/tests/integration_tests/chat_models/test_standard.py:
```diff
@@ -22,10 +21,6 @@ def chat_model_params(self) -> dict:
     def standard_chat_model_params(self) -> dict:
         return {"temperature": 0, "max_tokens": 100}
 
-    @pytest.mark.xfail(reason="Not implemented.")
-    def test_stop_sequence(self, model: BaseChatModel) -> None:
-        super().test_stop_sequence(model)
```
The `test_stop_sequence` integ tests should be kept, I verified that they're now working with your fix. Just replace the `@pytest.mark.xfail(reason="Not implemented.")` decorator with ` @pytest.mark.scheduled`

michaelnchin commented on libs/aws/tests/integration_tests/chat_models/test_standard.py:
```diff
@@ -52,7 +48,3 @@ def standard_chat_model_params(self) -> dict:
     def supports_image_inputs(self) -> bool:
         return True
-
-    @pytest.mark.xfail(reason="Not implemented.")
-    def test_stop_sequence(self, model: BaseChatModel) -> None:
-        super().test_stop_sequence(model)
```
Same as above, keep the test and replace the `xfail` decorator.

michaelnchin reviewed (changes requested):
Implementation LGTM! Few comments on the tests.

zhaoJoseph commented:
Thank you again for the quick response! I was considering removing the test case from `test_standard.py` because when running the integration tests I got an overrided failure `FAILED tests/integration_tests/chat_models/test_standard.py::TestBedrockStandard::test_no_overrides_DO_NOT_OVERRIDE - AssertionError: Standard tests overridden without @pytest.mark.xfail(reason="..."): ['test_stop_sequence']
FAILED tests/integration_tests/chat_models/test_standard.py::TestBedrockUseConverseStandard::test_no_overrides_DO_NOT_OVERRIDE - AssertionError: Standard tests overridden without @pytest.mark.xfail(reason="..."): ['test_stop_sequence']`
Should we ignore this result?

michaelnchin commented:
That makes sense @zhaoJoseph - revised my suggestions, let's stick with removing the `xfail` tests.

3coins reviewed (approved):
LGTM! Thanks for submitting this change.
